addons/osfstorage/models.py

```diff
@@ -307,7 +307,8 @@ def create_version(self, creator, location, metadata=None):
         version._find_matching_archive(save=False)
 
         version.save()
-        self.versions.add(version)
+        # Adds version to the list of file versions - using custom through table
```

Comment:
The file's `target_object_id` would be very useful here as well. If it were all on one index, I think the order would be `provider`->`target_content_type_id`->`target_object_id`, though we'd then probably need one that's `provider`->`target_content_type_id`->`id`, which might be excessive. We might want to chat options on this.

Reply:
I'm not sure I understand why `provider` or `target_content_type_id` would go on the through table instead of the BaseFileNode object (where these fields are already stored). 
If we move the file to another location, say a folder, like you alluded to, the BaseFileNode's target_object_id changes, it's not a copy that's made. So we'd have to update the through table as well.

Reply:
Makes sense. Let's not do that, then.

Reply:
but I mean..that's what we're doing with the `name` now. The latest is on the BaseFileNode and then a history of the name is on each file-version combo in the through-table.  I wonder if we should take `name` off of BaseFileNode, then.

Reply:
I guess that depends on how we get the file information. If we ever just get it from the BaseFileNode, then it makes sense to hold the latest there rather than caching it, because that will be arguably the most commonly used piece of information for that file. But if we always go through the new through table, then removing it makes sense.

(An aside: github is acting weird with comments; I thought I had two copies of my previous comment, so I deleted one, and now there are none.)

Reply:
hmmm, on second thought files from other addons might not have versions, I don't know enough about this yet.

Reply:
ok, looked into this some, definitely possible and common that files don't have versions on addons so it seems like the name still needs to reside on the BaseFileNode.   If the files do have versions, we'll store the name associated on this custom through table as well.